SCRUM-1185 person, data provider, experimental conditions updates

chris-grove opened the pull request.

gildossantos commented on model/schema/phenotypeAndDiseaseAnnotation.yaml:
```diff
@@ -301,15 +302,16 @@ classes:
-      - condition_chemical                # ChemicalOntologyId in JSON schema
+      - unique_id                         # calculated centrally at the Alliance after import from MODs
+      - condition_class                   # conditionClassId in frozen JSON schema
+      - condition_summary                 # conditionStatement in frozen JSON schema; for new DQM submissions should not be submitted but rather calculated centrally at the Alliance
```
Tests are currently failing because they're expecting "condition_summary" (required), but test files still have "condition_statement". Fixing files depends on resolution of the above question.

chris-grove commented:
@mluypaert @gildossantos 
I think I've addressed both of your points:

- I've reinstated the 'condition_statement' slot with some explanation added as to intended usage distinct from 'condition_summary'
- I've removed the requirement for 'condition_summary'

Please have a look and see if it satisfies your requirements.

chris-grove commented:
@markquintontulloch 
Please note that 'mod_id' has now been changed to 'mod_entity_id' and that the disease note slots have now been replaced by 'related_notes' to point to a list of Note objects. This will need to be updated in the WB test files. The Note class 'free_text', 'note_type' and 'private' boolean are all required. I think the note type for the 'disease_annotation_summary' replacement should be submitted as 'disease_summary'.

markquintontulloch reviewed (approved):
@chris-grove - I've pushed a couple of commits to this branch.  The first changes the WB test file to work with the modified schema, the second makes the 'name' slot of the VocabularyTerm class the identifier for this class.  Without an identifier, the class automatically gets inlined in the generation of the JSON schema - we don't want this as we don't want to have to submit the VocabularyTerm object, just the name.

chris-grove commented:
@markquintontulloch 
Fantastic, thanks Mark!
